packages/ckeditor5-table/src/tablecolumnresize/tablecolumnresizeutils.ts

```diff
@@ -0,0 +1,517 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module table/tablecolumnresize/tablecolumnresizeutils
+ */
+
+import { isEqual } from 'lodash-es';
+
+import { Plugin, type Editor } from 'ckeditor5/src/core.js';
+
+import type { DowncastWriter, Element, ViewElement, ViewNode } from 'ckeditor5/src/engine.js';
+import TableUtils from '../tableutils.js';
+
+import { COLUMN_MIN_WIDTH_IN_PIXELS } from './constants.js';
+import {
+	clamp, getColumnEdgesIndexes, getColumnGroupElement, getDomCellOuterWidth,
+	getElementWidthInPixels,
+	getTableColumnsWidths, getTableWidthInPixels, sumArray, toPrecision
+} from './utils.js';
+
+import TableWalker from '../tablewalker.js';
+
+/**
+ * The table column resize editing plugin.
+ */
+export default class TableColumnResizeUtils extends Plugin {
```

Comment:
I think this needs a bit of a cleanup.

There's already a `utils.ts` file with helpers dedicated for column resize feature. At the first glance, I don't see how functions added as methods of `TableColumnResizeUtils` plugin are different than those helpers in terms of semantics. So either all of them should be just helper functions not collected in a plugin, or all of them be kept in plugin. The situation where there's no clear distinction will raise confusion. Unless of course I don't see something, please educate me then 😄 

If you go with a plugin, please make it similar to `TableUtils` one - a real collection of utils and nothing more, so all the methods should be public and independent. Helpers for helpers can be kept outside of the plugin.

Reply:
I kept `TableColumnResizeUtils` and moved all private methods outside this plugin.